src/main.ts

```diff
@@ -57,6 +59,9 @@ export async function activate(context: vscode.ExtensionContext): Promise<IPower
     const settings = getSettings();
     logger.writeVerbose(`Loaded settings:\n${JSON.stringify(settings, undefined, 2)}`);
 
+    const RenameSymbol = new RenameSymbolFeature();
+    languageRenameProvider = vscode.languages.registerRenameProvider(documentSelector,RenameSymbol);
```

Comment:
This should be structured under the languageClientConsumers below and implemented as a feature. I know it is a bit obtuse but thats how the extension was originally architected and it enables features to be more pluggable.